### DISCUSSION TITLE
Having a community manager

### DISCUSSION THREAD

**Discussion by Piedone** [OrchardCMS/OrchardCore#14274] [Ideas]
Right now, Orchard Core doesn't have any specific marketing to the outside world: people share it via word of mouth, there are stray blog posts and talks here and there (though recently I've done that [more methodically](https://github.com/OrchardCMS/OrchardCore/discussions/14191)), but nobody is really doing outreach and marketing as one does for a product. Similarly, there’s no kind of product owner who’d oversee the project from a more holistic view.

Now, Orchard is more than a product in the commercial sense, not in the least it being a community-driven open-source software, of course, but still, it needs more work done to spread the word and to improve. This I think needs someone, let's call them Community Manager, who does this while not yet full-time, but at least on some level continuously.

I'd imagine they'd do e.g. these, but feedback on this and everything else here is welcome:

- Prefiltering and perhaps responding first to some issues (e.g. to ask for details before a developer takes on it), moderating community activities.
- Proposing new features and directions of development to the development team, based on feedback and discussions around Orchard (GitHub, Gitter, Discord, elsewhere) and trends and news in the rest of the world.
- Taking on [ecosystem tasks](https://github.com/OrchardCMS/OrchardCore/issues?q=is%3Aopen+is%3Aissue+label%3AEcosystem), including organizing events like Harvest.
- Evangelizing Orchard Core in the .NET community to reach out to developers, and in other communities to reach out to potential users and decision-makers. Spread the word in technologies OC uses like [it was the case with ZAP](https://www.zaproxy.org/blog/2023-12-08-automated-zap-scans-for-orchard-core-apps/), e.g. Azure, AWS, Elasticsearch, Lucene, OpenID, Mini Profiler... Building the Orchard Core brand.
- Writing user-level documentation.
- Perhaps [doing releases](https://docs.orchardcore.net/en/latest/docs/topics/publishing-releases/) too.
- [This article](https://opensource.com/article/21/9/community-manager) and [this other one](https://www.redhat.com/en/blog/many-faces-community-manager) have good further collections, and [The Open Source Way 2.0](https://www.theopensourceway.org/the_open_source_way-guidebook-2.0.html) looks like a helpful resource for our would-be Community Manager.

This person I think should possess the following skills and interests:

- Enthusiastic about open-source software and the web.
- Passionate to work with the people of open-source communities.
- Not necessarily a developer (but some level of development skills would be beneficial), but a good understanding of how a CMS and specifically Orchard Core works (the latter can be acquired after joining, of course), in which kind of technological context it can be used, and overall be a geek. They also need to be comfortable working with GitHub, Markdown, Git.
- Good with marketing, sales, copywriting (including documentation), and communication in general related to open-source software.
- Ability to help design software projects. I.e., distill user needs and create specifications (e.g. user stories, wireframes) for concrete features.

I’d emphasize that I don’t want to put a boss on top of the contributors of Orchard Core. I don’t want anybody to tell volunteer contributors who work on Orchard in their free time what they should work on. Rather, this would be a person to get more people to join us and work with and on Orchard, shepherd the project, foster the community, help contributors, and do work for Orchard that needs to be done but nobody is keen on doing it right now.

The work of this person needs to be funded somehow. This can be done with a combination of funding provided by Lombiq, .NET Foundation (they still have some available from the last Harvest, we can at least bootstrap with that), and financial support by the community (GitHub or Open Collective sponsorship).

We can perhaps get in touch with [Eric Schultz](https://www.linkedin.com/in/wwahammy/), who was with Orchard in the very beginning as kind of like this, as the contact for the Outercurve Foundation before Orchard moved to .NET Foundation (who we don’t hear about too much related to Orchard), at least for some tips.

Now where to find such a person is a good question. Apart from bringing it up in the Orchard community, I’d post to LinkedIn.

What does everyone think?

**Comment by larremp** [comment]
Hi @Piedone, I agree with the "needs" you mentioned:
1. Marketing of OC
2. User-level documentation
3. Event Organizer

However, I think for these tasks to be done in an excellent way, it might be done by different persons (there must be focus and expertise). Also, when "salary" is involved, it becomes a difficult conversation. 

But maybe, these functions can still be done by volunteers--just an idea. But surely, the topic needs further serious discussion.

Larrem

**Comment by Piedone** [comment]
What we discussed today in addition to the above:

- Larrem suggested starting with volunteers, not paid people and more than one person take on these.
- Skeptical about the current community members wanting to do most of these voluntarily.
- Why don't we pay for somebody to work on DI, or other coding tasks as well, then? It's just about different people liking to do different things.
- This is easier to ask volunteers to do if the task is smaller, i.e. divide this up, and one person is only responsible for only a few of these. And only for a set amount of time, like for a year, then we select new people (or renew).
- Someone needs to oversee the whole thing though, even if they don't do every single thing themselves, to keep to a specific direction.
- Look around how other projects do this, the ones that don't have a specific company behind them.

**Comment by larremp** [comment]
Upon further reflection, here are my additional suggestions/ideas:


[OCSuggestedRolesAndResponsibilities.docx](https://github.com/OrchardCMS/OrchardCore/files/12547904/OCSuggestedRolesAndResponsibilities.docx)

**Reply by Piedone** [reply]
Awesome suggestions! Adding them as Markdown below for accessibility.

Some notes:

- Marketing materials for community use: we can start with https://github.com/Lombiq/Orchard-Ambassadors-Toolbox.
- We had the [Orchard Ambassadors](http://ambassadors.orchardproject.net/) concept before. We could revive this.

---

My suggestions:

1. Define the role and responsibilities
  1. Marketing
    1. Create and execute a short-term (1-year) marketing plan
    2. Facilitate discussion for a long-term (3-5 years) marketing plan
    3. Facilitate discussion for a Gartner Quadrant [https://www.gartner.com/en/research/methodologies/magic-quadrants-research](https://www.gartner.com/en/research/methodologies/magic-quadrants-research)
    4. Promote Orchard Core
      1. Other Open Source projects/forums
      2. Tech Talks
      3. CMS-related events
    5. Suggest marketing improvements to the main OC website
    6. Create marketing materials for community use
    7. Create and manage marketing assets
  2. Events Organizer
    1. Harvest event
    2. Help manage and promote weekly Podcast/Forums/Meetings
    3. Create a quarterly event
    4. Create a yearly event
  3. User-level Documentation
    1. Create user-level documentation
    2. Possibly coordinate effort with other contributors
    3. Updated Quick-Start Guides
    4. Updated How-Tos
  4. Training Organizer
    1. Organize Free or Paid workshops
    2. Both for entry-level users and advanced developers
    3. Work with OC contributors or developers willing to help
    4. Organize a consistent training schedule
      1. Monthly
      2. Quarterly
2. Term Limits for the role
  1. 1 year terms
  2. Jan 1 through Dec 31
3. How to Select/Appoint volunteers
  1. These positions should be volunteers and NOT paid positions
  2. If more than 1 person want to volunteer to a position, let's create a voting system

**Comment by elaurentin** [comment]
4)
 iv. Promote OC over social media (YouTube, Discord, X)
 v. Partnership opportunities for OC developers to develop plugins, themes for OC

**Reply by Piedone** [reply]
Thank you! Can you specify a bit what you mean by partnership opportunities?

**Reply by elaurentin** [reply]
A market store for those who want to develop plugins, themes for OC. Just like in previous OrchardCMS.
I am sure many developers will be interested to develop plugins and themes. Thank you Piedone!

**Reply by Piedone** [reply]
I see, thank you. I'll start writing a docs page to summarize what we all gathered here.

**Reply by Piedone** [reply]
Also: https://github.com/OrchardCMS/OrchardCore/discussions/14504

**Comment by SzymonSel** [comment]
Nice, I like the enthusiasm! I'm remember back in the days, when we started doing some stuff around with Orchard v1. I think it was @stevetayloruk who took a lead on the matter. Perhaps he could share the "do's" and "don'ts"?

**Reply by Piedone** [reply]
Steve started an initiative for a new Orchard homepage back in the day, and he also blogged about it: https://www.stevetaylor.me.uk/the-new-orchard-cms-website/

**Comment by Piedone** [comment]
Related issues:

- https://github.com/OrchardCMS/OrchardCore/issues/15029
- https://github.com/OrchardCMS/OrchardCore/issues/15034

I also started to go through old PRs and clean them up, closing what's not applicable anymore, fixing what was easy, asking the author otherwise.

**Comment by Piedone** [comment]
Related discussion: https://github.com/OrchardCMS/OrchardCore/discussions/14504

**Comment by viktoriamagyar** [comment]
Hi everyone!

My name is Viktória, I am a business developer at Lombiq Technologies.
I've been talking with Zoltán and Benedek for a while now, about taking on the position of Community Manager for part of my working time, as far as my knowledge and experience allows.
What I am most interested in and feel I could take on responsibility for: (I collected these from previous messages and links)

- manage the interactions between the various participants, moderate awkward (or downright aggressive) communications, help encourage new members to contribute, raise the visibility of the project into new areas or market sectors, and all the other pieces that go into keeping a project healthy. This means that the community manager will hopefully have some support from at least one set of contributors but will still need to build consensus across the rest of the community.
- Marketing/outreach – this is about raising the visibility of the project, either in areas where it is already known or new markets/sectors. There are lots of sub-tasks such as branding, swag ordering (and distribution!), analyst, and press relations.
- Event management – setting up meetups, hackathons, booths at larger events or, for really big projects, organising conferences.
- Community lubrication – this is about finding ways to keep community members talking to each other, celebrate successes, mourn losses, and generally keep conversations civil at least and enthusiastically friendly at best.
- Partner management – as a project grows, partners (open source projects, academic institutions, charities, industry consortia, government departments, or commercial organisations) may wish to be associated with the project. Managing expectations and understanding the benefits (or dangers) and relative value can be complex and time-consuming, and the community manager is likely to be the first person involved.
- it needs more work done to spread the word and to improve.
- Based on the link you sent these are the things that I am interested in and would be capable of doing:
- Marketing of OC
- Prefiltering and perhaps responding first to some issues (e.g. to ask for details before a developer takes on it), moderating community activities.
- Event Organizer
- Promote Orchard Core
- Training Organizer
- Promote OC over social media (YouTube, Discord, X)
- User-level Documentation - Not sure about that. If you think I have the knowledge for that I am open, but to rather smaller tasks, the others are much more interesting.

**Reply by viktoriamagyar** [reply]
Please share with me your opinion!

**Reply by Skrypt** [reply]
What's your opinion about the Orchard Core brand itself? Should we try push it more by trying to put marketing around it or simply push it through partners like Lombiq does?

**Reply by Skrypt** [reply]
Just like adding Lombiq Dojo course content in the Orchard Core documentation like proposed here: https://github.com/OrchardCMS/OrchardCore/pull/15475

I don't mind, just asking. Because I think we could easily push the OC brand more too.

**Reply by viktoriamagyar** [reply]
Thank you for your question! :) 
It depends on what exactly we want to do.
I think that by giving Orchard Core a bigger marketing presence, it should result in an increase in partner inflow by default. 
In my opinion, what would be most needed in any online presence is to have information and displays about the technology and services available in everyday language. Something that can grab the attention of a person without technological knowledge. In my opinion, much more visibility could be achieved with a well thought out and planned marketing strategy, this is very important!
As long as someone inexpert is looking for an IT solution to their problems, I think this is the shortest way to get there. What also matters is how quickly and how you find someone.

**Reply by Skrypt** [reply]
Yeah, one of the issues with Orchard Core is that it is not as accessible as other CMS's out there. For different reasons.
Also, some CMS's out there are backed up with commercial products. One that I like to take a look at is Ghost https://ghost.org/ as I really like how they approach the blog UI. This is an open source project backed up by a commercial product which has a lot documentation / marketing pages on their website to promote both the commercial services and the tool itself.

Now, if we compare with what we have on Orchard website we are missing a lot.

But maybe we should only start by trying to explain what is the advantage of using Orchard Core vs other C# similar projects first. Make a respectful comparison with other CMS like Umbraco as an example. Find the strength points in using orchard then try to find marketing around it.

What is Orchard Core for is the question. :smile:

**Reply by hishamco** [reply]
@Skrypt the UI is a lacking part in OC coz everyone wants his/her styles :) I see many CMSs have a pretty look and feel, that's why one of the points I suggest for OC 2.0.0 is UI Refreshment

**Reply by Skrypt** [reply]
I don't think styles (CSS) and UI is related here @hishamco 
Of course we use Bootstrap and we need some basic framework to save ourselves time.
But, yes, UI implementations are missing. Like using a right toggling panel for some actions. Adding some Vue or Blazor apps to handle these things. Of course if we had a new admin UI based on Blazor that be a strong seller point for new tech guys but at the same time MVC is there to stay still.

Alse the Blazor hype is a dev thing. People that wants to have a website are generally not developpers; they are clients who are trying to find which CMS they should use and I think that's the marketing that is missing for OC.

**Comment by larremp** [comment]
Hi @viktoriamagyar, here's my suggestion in terms of areas of focus as a community manager at this point in time in the evolution of Orchard Core:

1. Create and maintain a repository of relevant marketing materials for OC.
- The repository should be easy to access and search by all.
- Show a list of sites using OC.
- Show current performance benchmarks.
- Show "Why consider Orchard Core?"

2.  Organize a regular workshops/seminars.
- 101 workshops for new users considering or poking around OC.
- Advanced workshops for developers.
- Paid or free.
- Create a quarterly schedule.
- Run it regularly so there's consistency.

3. Organizer yearly Harvest.
- Start small, maybe at least 10 attendees (as suggested by @sebastienros .
- If it is still possible, let's do one this year (2024).
- Let's make this a consistent yearly activity.

Thanks to @Piedone for working hard to make this happen!

**Reply by Piedone** [reply]
Harvest organization is going on FYI: https://github.com/OrchardCMS/OrchardCore/discussions/15260

**Reply by viktoriamagyar** [reply]
Thank you @larremp! 
As we talked about in the meeting these tasks would be the closest to me. And as you said I can take responsibility for this amount of tasks as a part-time Community manager.

**Comment by Piedone** [comment]
In the last few months, I've gone through all open pull requests, to only keep active ones, and merge everything that's worthwhile. Going forward, I can do PR management like this. It's not that I'd review everything, but rather, make sure that we don't keep stale PRs forever, and that contributors get feedback quickly.

See https://github.com/OrchardCMS/OrchardCore/issues/15029 for further ideas.

**Comment by csrcd** [comment]
I find this discussion very insightful 👏. In an open-source project like Orchard Core, technical excellence alone isn’t enough—without visibility, structured community engagement, and effective communication, even great solutions can remain under the radar. The idea of having a Community Manager, whether part-time or distributed across several contributors, seems like a strong step toward more strategic and sustainable growth.

Nowadays, many successful projects stand out thanks to how well they organize their ecosystem—documentation, onboarding experience, marketing, and events. It’s often an underestimated lever, yet crucial for attracting both developers and non-technical users.

Personally, I’m very interested in topics like project structuring, digital visibility, and building ecosystems around tech initiatives. I’ve started sharing some thoughts and resources on this here: https://csrcd.github.io

Looking forward to seeing how this evolves within the community 🚀[.](https://csrcd.github.io)
